automation/orchestrator.go

```diff
@@ -349,6 +349,18 @@ func (o *Orchestrator) listenForCancelationAndUnlock(ctx context.Context, cancel
 }
 
 func (o *Orchestrator) runWorkflow(ctx context.Context, wf *workflow.Workflow, runID string) error {
+	go func() {
+		if err := o.bus.Publish(ctx, event.ETAutomationRunQueuePush, &runID); err != nil {
+			log.Debug(err)
+		}
+	}()
+	defer func() {
```

Comment:
Unsure about how this flies, especially with ctx cancelations. Also real ugly, but expect you to pull this into the queue interface soon.

Reply:
So "Push" just means it goes on the queue, and "Pop" just means we've pulled it off the queue (not that the entire run as finished). With that in mind, "Push" is in the right place, but "Pop" should go right after the "lock" (and not in a defer), since that is what is simulating "coming off the queue" at the moment.

Reply:
ahhh, right right! ty